### PR TITLE
Sqlean Time extension

### PR DESCRIPTION
**Pull request by pedrocarlo**
This PR implements a sqlean time compatible extension. I would appreciate some help to review my code and see if there are ways to enhance it. Also, if there is some edge case, I have missed please tell me.
https://github.com/nalgeon/sqlean/blob/main/docs/time.md

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -82,6 +82,10 @@ test-vector:
 	SQLITE_EXEC=$(SQLITE_EXEC) ./testing/vector.test
 .PHONY: test-vector
 
+test-time:
```
Comment (PThorpe92) [inline Makefile:85]: We should probably move this to the `test-extensions`, in `testing/extensions.py` 

Depending on whether or not @penberg wants this included in the default features, you would want to make sure it's building first so add it to this line:
```Makefile
test-extensions: limbo
	cargo build --package limbo_regexp --package limbo_time
 ```
you can add a function there that (optionally) loads it first.

EDIT: nvm I looked at how much it's actually testing, and rewriting all those in python certainly isn't necessary. So it just depends on whether it needs to be loaded first.

File: extensions/time/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+[lib]
+crate-type = ["cdylib", "lib"]
+
+[features]
+static = ["limbo_ext/static"]
+
+[dependencies]
+chrono = "0.4.39"
+limbo_ext = { path = "../core", features = ["static"] }
```
Comment (PThorpe92) [inline extensions/time/Cargo.toml:17]: In order for the extensions to work when loaded at runtime(without "static"), we need to add the following
```toml
[target.'cfg(not(target_family = "wasm"))'.dependencies]
mimalloc = { version = "*", default-features = false }
```

File: extensions/time/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,1019 @@
...
+    };
+    ($e:expr, $msg:expr) => {
+        match $e {
+            Some(val) => val,
+            None => return Value::error_with_message($msg.to_string()),
+        }
+    };
+}
+
+macro_rules! tri {
```
Comment (PThorpe92) [inline extensions/time/src/lib.rs:81]: love these! :tada:  I'm thinking maybe these should be added to and exported from `extensions/core`. Might have to change the names to be a bit more clear though.

Comment (pedrocarlo) [reply extensions/time/src/lib.rs:81]: Awesome thanks! I wanted the tri macro to be named try! but it would conflict with rust's now deprecated macro. Maybe we could  rename ok_tri to opt_try? And value_tri to value_check? I am a bit bad at naming macros, unfortunately.

File: extensions/time/Cargo.toml (hunk 2 above)
Comment (pedrocarlo) [reply extensions/time/Cargo.toml:17]: Yeah, I read it a couple of days ago in the extension core readme, but totally forgot to add it here. 
https://github.com/tursodatabase/limbo/tree/main/extensions/core.
Will correct this

File: Makefile (hunk 1 above)
Comment (pedrocarlo) [reply Makefile:85]: This is one thing, I was a bit uncertain on how to do properly. I really wanted to test using tcl like we do with other tests, but to do so it had to be bundled together with limbo. So I added it to default, so I could just ```make limbo``` and then open the cli and test my changes. Maybe we could have another make command to bundle particular extensions together?
